### PR TITLE
FIX: Add compatibility for bucket folder paths in migrate_to_s3 task

### PR DESCRIPTION
**Pull request by rishabhnambiar**
**Example** without bucket folder path

```text
DISCOURSE_S3_ACCESS_KEY_ID="" \
DISCOURSE_S3_SECRET_ACCESS_KEY="" \
DISCOURSE_S3_REGION="us-west-2" \
DISCOURSE_S3_BUCKET="rishabhmultisite" \
DISCOURSE_S3_CDN_URL="https://discourse-cdn-aws1.com/pocketgems" \
rake uploads:migrate_to_s3
```

screenshot from 2019-01-07 17-40-23

**Example** with bucket folder path

```text
DISCOURSE_S3_ACCESS_KEY_ID="" \
DISCOURSE_S3_SECRET_ACCESS_KEY="" \
DISCOURSE_S3_REGION="us-west-2" \
DISCOURSE_S3_BUCKET="rishabhmultisite/pocketgems" \
DISCOURSE_S3_CDN_URL="https://discourse-cdn-aws1.com/pocketgems" \
rake uploads:migrate_to_s3
```

screenshot from 2019-01-07 17-36-22

### CODE REVIEW

**Comment by ZogStriP** [comment]
I wonder if there's a way to extract that logic so that both these rake tasks and the s3_helper can use it instead of duplicating it?

**Comment by rishabhnambiar** [comment]
> I wonder if there's a way to extract that logic so that both these rake tasks and the s3_helper can use it instead of duplicating it?

Sure, that makes sense. Do you have anything specific technique in mind, maybe an s3 [module](https://ruby-doc.org/core-2.2.0/Module.html)? If not, I will try to come up with something else :+1:

**Comment by ZogStriP** [comment]
Nothing in mind, maybe just some class methods on the s3_helper class since it already has some?

File: lib/s3_helper.rb [hunk 1]
```diff
@@ -23,6 +23,10 @@ def initialize(s3_bucket_name, tombstone_prefix = '', options = {})
       end
   end
 
+  def self.get_bucket_and_folder_path(s3_bucket_name)
+    return s3_bucket_name.downcase.split("/".freeze, 2)
```
Comment (ZogStriP) [inline lib/s3_helper.rb]: You don't need the `return` in Ruby when it's the last statement ;)

Comment (rishabhnambiar) [reply lib/s3_helper.rb]: Wow, I did that again :man_facepalming: 
Fixed, thanks!

**Pull request merged**